### PR TITLE
[PyQt6] Build Bindings for Qt6 on CI

### PR DESCRIPTION
**Pull request by troopa81**
This is the final pull request in order to build Python bindings with Qt6.

In order to build, this PR integrates #55674. I'll rebase this one once #55674 is merged

Most of Python tests are blocked and would be fixed later

### CODE REVIEW

File: python/PyQt6/core/__init__.py.in [hunk 1]
```diff
@@ -92,11 +94,6 @@ QgsVectorLayer.NoMarker = Qgis.VertexMarkerType.NoMarker
 QgsVectorLayer.NoMarker.is_monkey_patched = True
 QgsVectorLayer.NoMarker.__doc__ = "No marker"
 
-QgsSymbol.RenderHints = Qgis.SymbolRenderHints
```
Comment (nyalldawson) [inline python/PyQt6/core/__init__.py.in:95]: Why are these removed? Are they duplicates?

Comment (troopa81) [reply python/PyQt6/core/__init__.py.in:95]: They are generated by the new sipify, for instance, you got these for `Qgis.SymbolRenderHints`

```
Qgis.SymbolRenderHints = lambda flags=0: Qgis.SymbolRenderHint(flags)
QgsSymbol.RenderHints = Qgis.SymbolRenderHints
```

**Review by nyalldawson — APPROVED** [review]
Fantastic work !!
